sharedPlugins/inventoryImports/interfacingFunctions.js

```diff
@@ -141,7 +141,7 @@ function pollInventories(outputFile){
 	//return "/silent-command "+'local a="{"for b,c in pairs(game.players)do if c.connected then local d=game.players[b].get_inventory(defines.inventory.player_main).get_contents()local e=game.players[b].get_quickbar().get_contents()local f={}for g=1,game.players[b].force.character_logistic_slot_count do f[g]=game.players[b].character.get_request_slot(g)end;a=a..b..":{inventory:{"for h,i in pairs(d)do a=a.."['+"'"+'"..h.."'+"'"+']:"..i..","end;for h,i in pairs(e)do a=a.."['+"'"+'"..h.."'+"'"+']:"..i..","end;a=a.."},requestSlots:{"for h,i in pairs(f)do a=a.."['+"'"+'"..i["name"].."'+"'"+']:"..i["count"]..", "end;a=a.."}},"end end;a=a.."}"game.write_file("'+outputFile+'",a)'
-	return '/silent-command '+'local a='+"'"+'{"players":{'+"'"+'local b=false;for c,d in pairs(game.players)do if d.connected and d.character then if b then a=a..'+"'"+','+"'"+'else b=true end;local e=game.players[c].get_inventory(defines.inventory.player_main).get_contents()local f=game.players[c].get_quickbar().get_contents()local g={}for h=1,game.players[c].force.character_logistic_slot_count do g[h]=game.players[c].character.get_request_slot(h)end;a=a..'+"'"+'"'+"'"+'..c..'+"'"+'":{"inventory":{'+"'"+'local i=false;for j,k in pairs(e)do if i then a=a..'+"'"+','+"'"+'else i=true end;a=a..'+"'"+'"'+"'"+'..j..'+"'"+'":'+"'"+'..k end;for j,k in pairs(f)do if i then a=a..'+"'"+','+"'"+'else i=true end;a=a..'+"'"+'"'+"'"+'..j..'+"'"+'":'+"'"+'..k end;a=a..'+"'"+'},"requestSlots":{'+"'"+'i=false;for j,k in pairs(g)do if i then a=a..'+"'"+','+"'"+'else i=true end;a=a..'+"'"+'"'+"'"+'..k['+"'"+'name'+"'"+']..'+"'"+'":'+"'"+'..k['+"'"+'count'+"'"+']end;a=a..'+"'"+'}}'+"'"+'end end;a=a.."}}"game.write_file("'+outputFile+'",a,true,0)'
+	return '/silent-command '+'local a='+"'"+'{"players":{'+"'"+'local b=false;for c,d in pairs(game.players)do if d.connected and d.character then if b then a=a..'+"'"+','+"'"+'else b=true end;local e=game.players[c].get_inventory(defines.inventory.player_main).get_contents()local f=game.players[c].get_quickbar().get_contents()local g={}local z={}for j,k in pairs(e)do if z[j] ~= nil then z[j] = z[j] + k else z[j] = k end end for j,k in pairs(f)do if z[j] ~= nil then z[j] = z[j] + k else z[j] = k end end for h=1,game.players[c].force.character_logistic_slot_count do g[h]=game.players[c].character.get_request_slot(h)end;a=a..'+"'"+'"'+"'"+'..c..'+"'"+'":{"inventory":{'+"'"+'local i=false;for j,k in pairs(z)do if i then a=a..'+"'"+','+"'"+'else i=true end;a=a..'+"'"+'"'+"'"+'..j..'+"'"+'":'+"'"+'..k end;a=a..'+"'"+'},"requestSlots":{'+"'"+'i=false;for j,k in pairs(g)do if i then a=a..'+"'"+','+"'"+'else i=true end;a=a..'+"'"+'"'+"'"+'..k['+"'"+'name'+"'"+']..'+"'"+'":'+"'"+'..k['+"'"+'count'+"'"+']end;a=a..'+"'"+'}}'+"'"+'end end;a=a.."}}"game.write_file("'+outputFile+'",a,true,0)'
```

Comment:
- 🎨  [EN] This string is too long, must be refactored for maintainability
  - 🎨  [ES] Este string es demasiado largo, debe ser reconstruido para mantenibilidad
- 💡 [ES]  Se podría crear una función para crear un set de funciones para crear este string dado que el anterior es muy similar.

Reply:
I only fixed it. Here is a better readable version of the inner code I used to debug it:
```lua
local a='{"players":{'
local b=false;
for c,d in pairs(game.players)do 
    if d.connected and d.character then 
        if b then 
            a=a..','
        else 
            b=true 
        end;
        
        local e=game.players[c].get_inventory(defines.inventory.player_main).get_contents()
        local f=game.players[c].get_quickbar().get_contents()
        local g={}
        local z={}       
        for j,k in pairs(e)do 
            if z[j] ~= nil then
                z[j] = z[j] + k
            else
                z[j] = k
            end
        end
        for j,k in pairs(f)do 
            if z[j] ~= nil then
                z[j] = z[j] + k
            else
                z[j] = k
            end
        end
        
        
        for h=1,game.players[c].force.character_logistic_slot_count do 
            g[h]=game.players[c].character.get_request_slot(h)
        end
        a=a..'"'..c..'":{"inventory":{'
        
        local i=false;
        for j,k in pairs(z)do 
            if i then 
                a=a..','
            else 
                i=true 
            end;
            a=a..'"'..j..'":'..k 
        end
        a=a..'},"requestSlots":{'
        
        i=false;
        for j,k in pairs(g)do 
            if i then 
                a=a..','
            else 
                i=true 
            end;
            a=a..'"'..k['name']..'":'..k['count']
        end
        a=a..'}}'
    end 
end;
a=a.."}}"
game.write_file("inventoryImports.txt",a,true,0)
```